xtask/src/dist.rs

```diff
@@ -61,18 +66,28 @@ fn dist_server(nightly: bool) -> Result<()> {
         run!("cargo build --manifest-path ./crates/rust-analyzer/Cargo.toml --bin rust-analyzer --release")?;
     }
 
-    let (src, dst) = if cfg!(target_os = "linux") {
-        ("./target/release/rust-analyzer", "./dist/rust-analyzer-linux")
+    let (platform, ext) = if cfg!(target_os = "linux") {
+        ("linux", "")
     } else if cfg!(target_os = "windows") {
-        ("./target/release/rust-analyzer.exe", "./dist/rust-analyzer-windows.exe")
+        ("windows", ".exe")
     } else if cfg!(target_os = "macos") {
-        ("./target/release/rust-analyzer", "./dist/rust-analyzer-mac")
+        ("macos", "")
     } else {
         panic!("Unsupported OS")
     };
 
-    fs2::copy(src, dst)?;
+    let src = format!("./target/release/rust-analyzer{}", ext);
+    let dest = format!("./dist/rust-analyzer-{}.tar.gz", platform);
```

Comment:
```suggestion
    let dest = format!("./dist/rust-analyzer-{}.gz", platform);
```

It is not a tarball.

Reply:
Hmm, I need to recap on this :)